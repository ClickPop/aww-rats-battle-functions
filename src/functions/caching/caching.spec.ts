import request from 'supertest';
import { app } from '../..';

const agent = request.agent(app);

describe('/caching/clear', () => {
  test('should return cleared token count', async () => {
    const res = await agent.post('/caching/clear').set('authorization', 'test');
    expect(res.statusCode).toEqual(200);
    expect(res.body).toMatchSnapshot();
  });

  test('should error if given incorrect api key', async () => {
    const res = await agent
      .post('/caching/clear')
      .set('authorization', 'wrong');
    expect(res.statusCode).toEqual(401);
    expect(res.body).toMatchSnapshot();
  });

  test('should error if no api key is supplied', async () => {
    const res = await agent.post('/caching/clear');
    expect(res.statusCode).toEqual(401);
    expect(res.body).toMatchSnapshot();
  });
});
