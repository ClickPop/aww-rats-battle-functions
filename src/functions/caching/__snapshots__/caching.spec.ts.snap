// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/caching/clear should error if given incorrect api key 1`] = `
Object {
  "error": "invalid api key on request",
}
`;

exports[`/caching/clear should error if no api key is supplied 1`] = `
Object {
  "error": "invalid api key on request",
}
`;

exports[`/caching/clear should return cleared token count 1`] = `
Object {
  "affected_rows": 666,
}
`;
